aws/table_aws_workspaces_directory.go

```diff
@@ -0,0 +1,324 @@
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_workspaces_directory.listWorkspacesDirectories", "api_error", err)
+			return nil, err
+		}
+
+		for _, items := range output.Directories {
+			d.StreamListItem(ctx, items)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getWorkspacesDirectory(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	DirectoryId := d.EqualsQualString("directory_id")
+
+	// check if workspace id is empty
+	if DirectoryId == "" {
+		return nil, nil
+	}
+
+	// Create Session
+	svc, err := WorkspacesClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_workspaces_directory.getWorkspacesDirectory", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Build the params
+	params := &workspaces.DescribeWorkspaceDirectoriesInput{
+		DirectoryIds: []string{DirectoryId},
+	}
+
+	// Get call
+	data, err := svc.DescribeWorkspaceDirectories(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_workspaces_directory.getWorkspacesDirectory", "api_error", err)
+		return nil, err
+	}
+
+	if len(data.Directories) > 0 {
+		return data.Directories[0], nil
+	}
+
+	return nil, nil
+}
+
+func listWorkspacesDirectoriesTags(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	DirectoryId := h.Item.(types.WorkspaceDirectory).DirectoryId
+
+	// Create Session
+	svc, err := WorkspacesClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_workspaces_directory.listWorkspacesDirectoriesTags", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Build the params
+	params := &workspaces.DescribeTagsInput{
+		ResourceId: DirectoryId,
+	}
+
+	tags, err := svc.DescribeTags(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_workspaces_directory.listWorkspacesDirectoriesTags", "api_error", err)
+		return nil, err
+	}
+
+	return tags, nil
+}
+
+// https://docs.aws.amazon.com/workspaces/latest/adminguide/workspaces-access-control.html
+func getWorkspaceDirectoryArn(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getDirectoryArn")
+	region := d.EqualsQualString(matrixKeyRegion)
+	DirectoryId := h.Item.(types.WorkspaceDirectory).DirectoryId
+
+	commonData, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	commonColumnData := commonData.(*awsCommonColumnData)
+	arn := "arn:" + commonColumnData.Partition + ":workspaces:" + region + ":" + commonColumnData.AccountId + ":directory/" + *DirectoryId
```

Comment:
@madhushreeray30 can you please attach an official doc wherein the ARN format is mentioned?